photon-client/src/main/scala/com/linkedin/photon/ml/io/scopt/ScoptGameParametersParser.scala

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.io.scopt
+
+import scala.language.existentials
+
+import org.apache.hadoop.fs.Path
+
+import com.linkedin.photon.ml.DataValidationType
+import com.linkedin.photon.ml.DataValidationType.DataValidationType
+import com.linkedin.photon.ml.Types.FeatureShardId
+import com.linkedin.photon.ml.cli.game.GameDriver
+import com.linkedin.photon.ml.data.InputColumnsNames
+import com.linkedin.photon.ml.evaluation.EvaluatorType
+import com.linkedin.photon.ml.io.FeatureShardConfiguration
+import com.linkedin.photon.ml.util.{DateRange, DaysRange, PhotonLogger}
+
+/**
+ * Base trait for any [[ScoptParser]] for a class derived from the [[GameDriver]].
+ */
+trait ScoptGameParametersParser extends ScoptParser {
```

Comment:
@joshvfleming 
@fastier-li 
@Adama94

Base class for the training and (soon to be) scoring parsers. Defines Scopt parameters for the shared parameters of the `GameDriver`.